components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/RequestObjectValidatorImpl.java

```diff
@@ -199,55 +205,21 @@ protected boolean validateAudience(String currentAudience, List<String> audience
     }
 
     /**
-     * Get the X509CredentialImpl object for a particular tenant and alias
+     * Get the X509Certificate object containing the public key of the OAuth client.
      *
-     * @param tenantDomain tenant domain of the issuer
-     * @param alias        alias of cert
-     * @return X509Certificate object containing the public certificate in the primary keystore of the tenantDOmain
-     * with alias
+     * @param clientId clientID of the OAuth client (Service Provider).
+     * @param tenantDomain tenant domain of Service Provider.
+     * @return X509Certificate object containing the public certificate of the Service Provider.
      */
-    protected Certificate getCertificateForAlias(String tenantDomain, String alias) throws RequestObjectException {
-
-        int tenantId;
-        String error = "Unable to Validate the Signature of Request Object";
-        tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
-        KeyStoreManager keyStoreManager;
-        // get an instance of the corresponding Key Store Manager instance
-        keyStoreManager = KeyStoreManager.getInstance(tenantId);
-        KeyStore keyStore;
+    protected Certificate getPublicCertOfOAuthApp(String clientId, String tenantDomain) throws RequestObjectException {
```

Comment:
since certificate is public, "Public" word seems redundant here

Reply:
Also is it safe to remove/rename this method? since this is a protected method there can be extensions.

Reply:
The idea behind removing this method was that it will no longer be useful since we are moving away from keystore based approach for storing certs. Anyways I agree the right way is to first deprecate and remove the protected method. 

Will reintroduce the method and deprecate it.